pkg/asset/machines/master.go

```diff
@@ -220,7 +221,9 @@ func (m *Master) Generate(dependencies asset.Parents) error {
 					mpool.Zones = append(mpool.Zones, zone)
 				}
 			} else {
-				mpool.Zones, err = installConfig.AWS.AvailabilityZones(ctx)
+				ctx2, cancel := context.WithTimeout(ctx, 1*time.Minute)
```

Comment:
I would rather not put the timeout in this call. Putting it here is somewhat fragile because it makes an assumption that this is going to be the very first call to the service endpoint. I would rather make a separate call where we are explicitly testing the accessibility of the service endpoints. Conveniently, we already have a function where we validate the service endpoints, namely `validateServiceEndpoints`.